NOTICE.txt

```diff
@@ -5556,40 +4296,8 @@ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
 THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
---------------------------------------------------------------------
-Dependency: github.com/shirou/w32
-Revision: bb4de0191aa41b5507caa14b0650cdbddcd9280b
-License type (autodetected): BSD-2-Clause
-./vendor/github.com/shirou/w32/LICENSE:
---------------------------------------------------------------------
-Copyright (c) 2010-2012 The w32 Authors. All rights reserved.
-
-Redistribution and use in source and binary forms, with or without
-modification, are permitted provided that the following conditions
-are met:
-1. Redistributions of source code must retain the above copyright
-   notice, this list of conditions and the following disclaimer.
-2. Redistributions in binary form must reproduce the above copyright
-   notice, this list of conditions and the following disclaimer in the
-   documentation and/or other materials provided with the distribution.
-3. The names of the authors may not be used to endorse or promote products
-   derived from this software without specific prior written permission.
-
-THIS SOFTWARE IS PROVIDED BY THE AUTHORS ``AS IS'' AND ANY EXPRESS OR
-IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
-OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
-IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY DIRECT, INDIRECT,
-INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
-NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
-DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
-THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
-THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-
 --------------------------------------------------------------------
 Dependency: github.com/Shopify/sarama
-Version: v1.24.1-elastic
-Revision: 355d120d09706d85bc6d2a84287be7fbd13ef453
```

Comment:
missing version or revision

Reply:
Where both version and revision is missing we are using our own forks for a dependency. How should we treat these cases? Should we remove these and add our own forks?

Reply:
Some overwrites seem to be older. Best would be if we can get rid of them by updating dependencies (if possible). Each rewrite adds potential maintenance overhead.

Go mod files can contain revision/version information as well. Can we still extract the version/revision from the original via the go.mod file? Maybe we can add fields like:

```
Dependency: github.com/Shopify/sarama
Version: v1.24.1  // <- original dependency version
Overwrite: ... repo with our fork
Overwrite-Revision: ...
```